pilot/pkg/xds/endpoint_builder.go

```diff
@@ -78,9 +82,18 @@ func (b EndpointBuilder) Key() string {
 	if b.service != nil {
 		params = append(params, string(b.service.Hostname)+"/"+b.service.Attributes.Namespace)
 	}
+	if b.networkView != nil {
+		for nw := range b.networkView {
+			params = append(params, nw)
+		}
+	}
 	return strings.Join(params, "~")
 }
 
+func (b *EndpointBuilder) IsMultinetwork() bool {
+	return b.push.Networks != nil && len(b.push.Networks.Networks) > 0
```

Comment:
what does this mean? if endpoints have network setting but the global meshNetworks map is not present? How do we distinguish between a temporarily unavailable meshNetworks map (one where the admin forgot to add it) vs one that is truly absent (i.e. user telling us to ignore networks field)?

Reply:
This just moves current behavior into a method. The name isn't great though, I'll try to clarify. 

I don't think we care about whether or not the user _meant_ to have the meshNetworks settings missing. But I guess it should go in the cache key so we can recompute if it's added/removed later.